lib/zoho_auth_client.rb

```diff
@@ -0,0 +1,40 @@
+class ZohoAuthClient
+  ACCESS_TOKEN_REQUEST_ENDPOINT = 'https://accounts.zoho.com/oauth/v2/token'.freeze
+
+  CLIENT_ID = ArchiveConfig.ZOHO_CLIENT_ID
+  CLIENT_SECRET = ArchiveConfig.ZOHO_CLIENT_SECRET
+  ZOHO_REFRESH_TOKEN = ArchiveConfig.ZOHO_REFRESH_TOKEN
+
+  # TODO: Can we get rid of this? We don't redirect, but it has to match
+  # whatever was filled in when creating the original grant_code
+  REDIRECT_URI = 'https://archiveofourown.org/support'.freeze
+
+  # TODO: Can we reduce this?
+  SCOPE = 'Desk.tickets.ALL,Desk.contacts.READ,Desk.contacts.WRITE,Desk.contacts.CREATE,Desk.basic.READ,Desk.search.READ'.freeze
+
+  def initialize
+    @access_token = access_token
```

Comment:
What's the purpose of assigning this to `@access_token`? The variable doesn't appear to be accessed anywhere.

(Also, doesn't this mean that calling `ZohoAuthClient.new.access_token`, as you do above, would cause the token to be retrieved/refreshed twice? Once when `access_token` is called in the constructor, and again when `access_token` is called on the resulting object?)

Reply:
Doh! Good catch.